transport/http/outbound.go

```diff
@@ -323,6 +339,13 @@ func (o *Outbound) withOpentracingSpan(ctx context.Context, req *http.Request, t
 }
 
 func (o *Outbound) withCoreHeaders(req *http.Request, treq *transport.Request, ttl time.Duration) *http.Request {
+	// Add default headers to all requests.
+	for k, vs := range o.headers {
+		for _, v := range vs {
+			req.Header.Add(k, v)
```

Comment:
Should we blow up if they try to use one of the core headers below? That might be better than having whatever they set be overwritten, for example.